hs/src/Reach/Simulator/Server.hs

```diff
@@ -270,6 +270,25 @@ getProgState sid = do
     Nothing -> return Nothing
     Just st -> return $ Just st
 
+checkIfValIType :: IType -> Bool
+checkIfValIType = \case
+  IT_Val _ -> True
+  IT_Fun _ _ -> False
+  IT_UDFun _ -> False
+
+initVals :: InteractEnv -> WebM (M.Map String String)
+initVals (InteractEnv iv'') = do
+  let iv' = M.toList iv''
+  let iv = filter (\(_,itype) -> checkIfValIType itype) iv'
+  return $ M.fromList $ map (\(var,itype) -> (var, show itype)) iv
```

Comment:
This function could be implemented entirely inside `M` with `M.filter checkIfValIType` and `M.map show`, which should be more efficient than going in and out of lists